tools/import-strings.sh

```diff
@@ -18,29 +18,445 @@
 #  $ git push
 #  $ gh pr create # (or manually)
 #
-# For the bigger picture on string import, export see the wiki.
+# For the bigger picture on string import, import see the wiki.
 #
-#  https://github.com/mozilla-mobile/focus-ios/wiki/Importing-and-Exporting-Strings
+#  https://github.com/mozilla-mobile/focus-ios/wiki/Importing-and-importing-Strings
 #
 
-set -e
+# -E: If set, any trap on ERR is inherited by shell functions, command substitutions, and commands executed in a subshell environment. The ERR trap is normally not inherited in such cases.
+# -e: Exit immediately if a command exits with a non-zero status.
+# -u: Treat unset variables as an error when substituting.
+# -o pipefail: The return value of a pipeline is the status of the last command to exit with a non-zero status, or zero if no command exited with a non-zero status
+set -Eeuo pipefail
 
-if [ ! -d Blockzilla.xcodeproj ]; then
-  echo "[E] Run this script from the project root as tools/export-strings.sh"
+# shellcheck disable=SC2155
+declare -r script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+# shellcheck disable=SC2155
+declare -r root_dir=$(cd "${script_dir}/.." &>/dev/null && pwd -P)
+
+[[ "${BASH_VERSINFO[0]}" -ge 4 ]] || {
+  cat <<EOF
+  #############################################################################
+  This script requires bash version 4 or higher in order to support 
```

Comment:
We need to be sure github actions use this otherwise the action to automatically import strings will fail.

Reply:
I'll check. I think the newest version of Bash is 5.5. I was initially concerned about the requirement for Bash 4.0+ until I realized how old it is. There are quite a few security issues with pre 4.0, and there might even be some more issues I'm not aware of that would require an update to 5.0+, but I'm not sure.

I'd imagine github actions is using the most up-to-date version, but I shouldn't make any assumptions - good call.

Reply:
> I'd imagine github actions is using the most up-to-date version

I would expect that to be 3.5 on a macOS image

```
$ /bin/bash --version
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin21)
Copyright (C) 2007 Free Software Foundation, Inc.
```